frame/contracts/src/exec.rs

```diff
@@ -537,13 +563,16 @@ where
 
 	fn instantiate(
 		&mut self,
-		code_hash: &CodeHash<T>,
+		code_hash: CodeHash<T>,
 		endowment: BalanceOf<T>,
 		gas_meter: &mut GasMeter<T>,
 		input_data: Vec<u8>,
 		salt: &[u8],
 	) -> Result<(AccountIdOf<T>, ExecReturnValue), ExecError> {
-		self.ctx.instantiate(endowment, gas_meter, code_hash, input_data, salt)
+		let executable = E::from_storage(code_hash, &self.ctx.config.schedule)?;
+		let result = self.ctx.instantiate(endowment, gas_meter, &executable, input_data, salt)?;
+		executable.store();
```

Comment:
why do we call `executable.store` after `self.ctx.instantiate` ? isn't this an error:
`ctx.instantiate` will collect rent of the new contract, but the `occupied_storage` will not take the new contract into account yet

Reply:
You are right. The rent collected is a bit too high here because the refcount is only incremented in `store`.  However, the first rent is only for a single block as a basic sanity check.

Nonetheless, I think a fix for this would be easy: Move the increment from `store` to `instantiate`. Let me try if this is really that easy.

Reply:
We must increment in in `store` because we cannot rely on the `recounter` that is stored inside the `Executable`. This can be wrong when the instantiated contract did instantiate the same contract in its constructor.

I think it is OK to collect the first "1 block rent" as if the current contract wasn't already contributing.

Reply:
Yes ok, maybe just a comment saying so, but I'm fine